src/kem/kyber/ntt.rs

```diff
@@ -17,6 +17,74 @@ const ZETAS_MONTGOMERY_DOMAIN: [KyberFieldElement; 128] = [
     -1530, -1278, 794, -1510, -854, -870, 478, -108, -308, 996, 991, 958, -1460, 1522, 1628,
 ];
 
+// Over time, all invocations of ntt_representation() will be replaced by
+// invocations to this function, upon which this function will be renamed back to
+// ntt_representation().
+#[inline(always)]
+pub(in crate::kem::kyber) fn ntt_with_debug_asserts(
```

Comment:
I don't think `coefficient_bound` should be an argument here.
This makes things look pretty weird and requires this additional function for now.
It looks this value should always be static, so it could be a const generic.
So it's probably something that requires some hax support to be nice in the end. But until then maybe a const generic is the way to go?
The remaining function that still uses the oldversion, could get `usize::MAX - 319584` to avoid having two functions. But then the reduction actually fails. So looks likes that's not all correct yet.

The other question is whether we need the strict individual bound or if a slightly larger, but generic bound is enough.

Reply:
ok, so let's get this in and then do a follow up.
We should make sure to get the F* extract to at least lax type check this week to make sure whatever we do here actually makes sense for the proofs.